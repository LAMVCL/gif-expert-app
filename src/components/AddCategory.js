import React, { useState } from 'react';
import PropTypes from 'prop-types';

//Desestructuramos el prop mediante llaves {}
export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');
    const handleInputChange = (e) => {
        setInputValue(e.target.value);

    }
    const handleSubmit = (e) =>{
        //Bloqueamos la actualización de la página
        e.preventDefault();

        //Validamos que el input posea texto mediante un length
        if (inputValue.trim().length > 2){
            //De ser valido asignamos el texto al array mediante setCategories
            setCategories(newCategorie => [inputValue,...newCategorie ]);
            //Vaciamos el input para evitar duplicados al presionar doble submit
            setInputValue('');
        }
        
    }
    return (
        <form className='formulario' onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input 
            type="text"
            value={inputValue}
            placeholder='Buscar...'
            onChange={handleInputChange}
            />
            <button ><i className="fas fa-search"></i></button>
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
