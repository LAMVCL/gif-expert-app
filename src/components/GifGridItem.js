import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({title, url}) => {

    return (
        <div className="card animate__animated animate__fadeInDown">
            <p>{title}</p>
            <img src={url} alt={title}></img>
        </div>
    )
}

GifGridItem.propTypes = {

    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
};
