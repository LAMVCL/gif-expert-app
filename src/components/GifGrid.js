import React, { useState, useEffect } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

    const [images, setImages] = useState([]);

    useEffect(() => {
        getGifs();
    }, [])

    const getGifs = async() =>{

        const urlFetch = `https://api.giphy.com/v1/gifs/search?q=${category}&limit=10&api_key=g1dCiL4mecdpaOWgTGtlvZzrLr0c9nBW`;
        const respuesta = await fetch(urlFetch);
        //Destructuramos data para evitar poner data.data
        const {data} = await respuesta.json();

        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images.downsized.url,
            }
        });

        console.log(gifs);
        //Le pasamos el array de 10 imagenes con su id, titulo y url
        setImages(gifs);
    }

    return (
        <>
            <h3>{category}</h3>
            <div className="card-grid">
                
                {
                    images.map( (img) => {

                        return <GifGridItem key={img.id} {...img}
                        />
                    })
                }
            </div>
        </>
        
    )
}
