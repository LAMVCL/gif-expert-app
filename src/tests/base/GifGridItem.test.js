import React from "react";
import { shallow } from "enzyme";
import {GifGridItem} from "../../components/GifGridItem";

describe('Pruebas en <GifGridItem/>', () =>{

    const title = 'Un titulo';
    const url = 'https://localhost/algo.jpg'

    const wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('Debe mostrar el <GifGridItem/> correctamente', () =>{

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de tener un pÃ¡rrafo con el title', () =>{

        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('Debe de tener la imagen igual al url y alt de los props', () =>{

        const img = wrapper.find('img');

        //Obtenemos props del atributo img.
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('Debe de tener animate__fadeInDown ', () => {
        
        const div = wrapper.find('div');
        const clase = div.prop('className');

        expect(clase.includes('animate__fadeInDown')).toBe(true);
    })
    
});