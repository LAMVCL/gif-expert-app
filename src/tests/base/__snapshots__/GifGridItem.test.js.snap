// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> Debe mostrar el <GifGridItem/> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <p>
    Un titulo
  </p>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
</div>
`;
